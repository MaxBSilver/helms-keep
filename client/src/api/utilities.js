export const host = 'http://localhost:5000/';

export const mockTask = {
  id: 1234,
  isCompleted: false, 
  message: 'Clean the kitchen'
}

export const mockNote = {
  id: 1234,
  title: 'My First Quest',
  tasks: [mockTask, mockTask]
}