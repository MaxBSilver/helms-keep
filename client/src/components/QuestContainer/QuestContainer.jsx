import React, { Component } from 'react';
import Quest from '../Quest/Quest';
import { connect } from 'react-redux';

export class QuestContainer extends Component {
  render() {
    const questCards = this.props.quests.length && this.props.quests.map(quest => <Quest data={quest} key={quest.id} />);
    
		return <section className="QuestContainer">{questCards}</section>;
	}
}

export const mapStateToProps = state => ({
  quests: state.quests
})

export default connect(mapStateToProps)(QuestContainer);
