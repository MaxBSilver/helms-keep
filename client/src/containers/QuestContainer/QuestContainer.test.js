import React from 'react';
import { shallow } from 'enzyme';
import { QuestContainer, mapStateToProps } from './QuestContainer';
import { mockNote } from '../../api/utilities';

describe('QuestContainer', () => {
	let wrapper;

	beforeEach(() => {
		wrapper = shallow(
		<QuestContainer 
			quests={[mockNote, mockNote]}
			results={null}
		/>);
	});

	it('should match snapshot', () => {
		expect(wrapper).toMatchSnapshot();
	});

	it('should match snapshot with results', () => {
		wrapper = shallow(
			<QuestContainer
				quests={[mockNote, mockNote]}
				results={[1234]}
			/>);
		expect(wrapper).toMatchSnapshot();
	});

	describe('MSTP', () => {
		it('should return an array of quests', () => {
			const state = { quests: [mockNote, mockNote], test: 'test' }
			const expected = { quests: [mockNote, mockNote] }
			const result = mapStateToProps(state);
			expect(result).toEqual(expected);
		});
	});
});
