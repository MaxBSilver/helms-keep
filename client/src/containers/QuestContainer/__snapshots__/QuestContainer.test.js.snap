// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuestContainer should match snapshot 1`] = `undefined`;
